### PR TITLE
Cleaned up json_data.class.php and brought to parity with XML. 

### PR DESCRIPTION
**Pull request by AshotN**
Some things still missing though, marked with TODO.

There may be some breaking changes.

### CODE REVIEW

**Comment by lachlan-00** [comment]
json isn't officially out yet but i'll run this against python3-ampache and see if it breaks anything

File: composer.json [hunk 1]
```diff
@@ -70,7 +70,8 @@
         "friendsofphp/php-cs-fixer": "~2.16.0",
         "kunalvarma05/dropbox-php-sdk": "^0.2.1",
         "tightenco/collect": "5.2.*",
-        "ampache/ampacheapi-php": "^1.0"
+        "ampache/ampacheapi-php": "^1.0",
+        "ext-json": "*"
```
Comment (lachlan-00) [inline composer.json:74]: i'll actually put this in the installer tests as well

Comment (AshotN) [reply composer.json:74]: PHPStorm had been complaining about it for a while now.

**Comment by AshotN** [comment]
Is there something holding this from being merged?

**Comment by lachlan-00** [comment]
> 
> 
> Is there something holding this from being merged?

Hours in the day! ;)

**Pull request merged**